proposals/FileSystemSubscriptionManager.md

```diff
@@ -0,0 +1,162 @@
+
+Hence, simply enabling it doesn't satisfy the goal. This approach might only work for websites that either can ensure a long-running service worker or are not really interested in using service worker registration for observing the file change events after the websites are closed.
+
+[fso]: https://github.com/whatwg/fs/blob/main/proposals/FileSystemObserver.md#handling-changes-made-outside-the-lifetime-of-a-filesystemobserver
+[chrome-sw]: https://developer.chrome.com/blog/longer-esw-lifetimes#background
+
+#### Outliving Service Worker
+
+What if updating FileSystemObserver to allow it to outlive without the limit of running in ServiceWorkerGlobalScope?
+
+If such an option is implemented, there needs to be mechanisms to tell when the browser should stop watching file changes.
+Handle file changes happen after the service worker is already terminated but the website is still open. Possibly needs a way to wake up a new service worker.
+
+There are existing mechanisms to auto wake up new service workers. Hence the next option.
+
+### Option 2. Utilizing Service Worker Registration
+
+The [ServiceWorkerRegistration][swr] interface represents registration of a service worker for a specific origin and scope. The browser maintains a persistent list of active ServiceWorkerRegistration even when the associated service worker is not actively running, and will wake up new service workers if a registered event happens.
+
+There is already existing example utilizing the interface to receive notifications.
+For example, `ServiceWorkerRegistration.pushManager` is the [PushManager] from the Push API that allows subscription to a push service.
+
+This doc below proposes a similar interface under ServiceWorkerRegistration to allow subscribing to file system change events.
+
+[swr]: https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration
+[PushManager]: https://developer.mozilla.org/en-US/docs/Web/API/PushManager
+
+## Proposed API
+
+### Web IDL
+
+First, define a new interface `FileSystemSubscriptionManager` that will be held under every ServiceWorkerRegistration, which manages subscriptions to changes to file systems:
+
+```webidl
+partial interface ServiceWorkerRegistration {
+  // Returns a reference to FileSystemSubscriptionManager interface, which allows
+  // for subscribing to specific file changes.
+  [SameObject] readonly attribute FileSystemSubscriptionManager fileSystem;
+};
+```
+
+It supports `subscribe()` and `unsubcribe()` to a `FileSystemHandle`, with [`FileSystemObserverObserveOptions`](https://whatpr.org/fs/165.html#dictdef-filesystemobserverobserveoptions).
+
+```webidl
+// Provides methods for managing file system subscriptions.
+interface FileSystemSubscriptionManager {
+  // Subscribes to changes to a FileSystemHandle with the browser with specific
+  // options. Returns a Promise that resolves when the subscription completes.
+  Promise<void> subscribe(FileSystemHandle handle,
```

Comment:
```suggestion
  Promise<undefined> subscribe(FileSystemHandle handle,
```

here and elsewhere. `void` is not valid IDL.

Reply:
Done.